chatkit/src/main/java/com/pusher/chatkit/messages/MessageService.kt

```diff
@@ -25,60 +25,66 @@ class MessageService(
     private val filesInstance get() = chatManager.filesInstance
 
     @JvmOverloads
-    fun messageEvents(messageLimit: Int? = null, callback: (Result<Message, Error>) -> Unit): Subscription {
-        val roomSubscription = RoomSubscription(room, chatManager.userStore, callback, chatManager)
-        with(chatManager) {
-            val path = when (messageLimit) {
-                null -> "/rooms/${room.id}?user_id=${currentUser.id}"
-                else -> "/rooms/${room.id}?user_id=${currentUser.id}&message_limit=$messageLimit"
-            }
-            return apiInstance.subscribeResuming(
+    fun messageEvents(
+        messageLimit: Int = 10,
+        callback: (Result<Message, Error>) -> Unit
+    ): Promise<Result<Subscription, Error>> = chatManager.currentUser
+        .flatMapResult { currentUser -> messageEventsPath(currentUser, messageLimit).asPromise() }
+        .mapResult { path ->
+            val roomSubscription = RoomSubscription(room, chatManager.userStore, callback, chatManager)
+            chatManager.apiInstance.subscribeResuming(
                 path = path,
-                tokenProvider = tokenProvider,
-                tokenParams = tokenParams,
+                tokenProvider = chatManager.tokenProvider,
+                tokenParams = chatManager.tokenParams,
                 listeners = roomSubscription.subscriptionListeners
             )
         }
+
+    private fun messageEventsPath(currentUser: CurrentUser, messageLimit: Int): Result<String, Error> = when {
+        messageLimit < 0 -> Errors.other("messageLimit should be greater than 0").asFailure()
+        else -> "/rooms/${room.id}?user_id=${currentUser.id}&message_limit=$messageLimit".asSuccess()
     }
 
+
     @JvmOverloads
     @UsesCoroutines
-    fun messageEvents(messageLimit: Int? = null): Promise<Result<Message, Error>> =
+    fun messageEvents(messageLimit: Int = 10): Promise<Result<Message, Error>> =
         Promise.promise {
-            messageEvents(messageLimit) { report(it) }
+            val subscription = messageEvents(messageLimit) { report(it) }
+            onCancel { subscription.cancel() }
         }
 
-    fun cursors(callback: (CursorsSubscription.Event) -> Unit) {
-        val cursorsSubscription = CursorsSubscription(currentUser, room, chatManager, callback)
-        with(chatManager) {
-            cursorsInstance.subscribeResuming(
-                path = "/cursors/0/rooms/${room.id}/",
-                tokenProvider = tokenProvider,
-                tokenParams = tokenParams,
-                listeners = cursorsSubscription.subscriptionListeners
-            )
-        }
-    }
+    // TODO: Create cursor service
```

Comment:
Is this intended to be left in here commented out?

Reply:
Yeah, need to create the cursor events, card coming up.